src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -302,7 +305,19 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary = False):
                 logging.debug("Updated job summary state history for job %s" % jobSummaryId)
 
             if job.get("fwjr", None):
-
+                cached_workflow = self.workflow_cache.setdefault(job['workflow'], {})
+                if job['task'] not in cached_workflow:
+                    workload = WMWorkloadHelper()
+                    workload.load(self.getWorkflowSpecDAO.execute(job['task'])[job['task']]['spec'])
```

Comment:
@ticoann @amaltaro 
some changes from failure to success those that are going the other way look to be coming from the fact the spec path is a dummy `'spec.xml` in the testing and it fails to load the spec at this line. So I think that otherwise this is ready unless you have some further comments/requirements?